homeassistant/components/intellifire/__init__.py

```diff
@@ -31,82 +42,152 @@
 ]
 
 
+def _construct_common_data(entry: ConfigEntry) -> IntelliFireCommonFireplaceData:
+    """Convert a config entry into IntelliFireCommonFireplaceData."""
+
+    return IntelliFireCommonFireplaceData(
+        auth_cookie=entry.data[CONF_AUTH_COOKIE],
+        user_id=entry.data[CONF_USER_ID],
+        web_client_id=entry.data[CONF_WEB_CLIENT_ID],
+        serial=entry.data[CONF_SERIAL],
+        api_key=entry.data[CONF_API_KEY],
+        ip_address=entry.data[CONF_IP_ADDRESS],
+        read_mode=entry.options[CONF_READ_MODE],
+        control_mode=entry.options[CONF_CONTROL_MODE],
+    )
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
```

Comment:
I see we are only adding data, do we really need to migate this or can we just add the data? Migrations are nice, but they break the integration when you revert to an older version.